src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -380,32 +359,38 @@ private PluginDescriptionFile findPluginDescription(Class<? extends JavaPlugin>
 	public void callEvent(Event event) throws IllegalStateException
 	{
 		events.add(event);
-		for (List<ListenerEntry> listeners : eventListeners.values())
-		{
-			for (ListenerEntry entry : listeners)
-			{
-				if (entry.isCompatibleFor(event))
-				{
-					entry.invokeUnsafe(event);
-				}
+		HandlerList handlers = event.getHandlers();
+		RegisteredListener[] listeners = handlers.getRegisteredListeners();
+		Arrays.stream(listeners).forEach(l ->callRegisteredListener(l,event));
+	}
+
+	private void callRegisteredListener(RegisteredListener registration, Event event){
+		if (!registration.getPlugin().isEnabled()) {
+			return;
+		}
+		try {
+			registration.callEvent(event);
+		} catch (AuthorNagException ex) {
+			Plugin plugin = registration.getPlugin();
+
+			if (plugin.isNaggable()) {
+				plugin.setNaggable(false);
+
+				server.getLogger().log(Level.SEVERE, String.format(
+						"Nag author(s): '%s' of '%s' about the following: %s",
+						plugin.getDescription().getAuthors(),
+						plugin.getDescription().getFullName(),
+						ex.getMessage()
+				));
 			}
```

Comment:
I would suggest making a `nag(Plugin plugin, String reason)` method here.
It seems like this may come in useful at another time too.

Reply:
TO be honest this code is pretty much C+P'd from the Spigot source, tweaked to support a testing environ.  Again nagging can probably just be removed for testing purposes.  I mean for CI purposes logging is a waste of time.